sql/plan/flush.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/grant_tables"
+)
+
+// FlushPrivileges reads privileges from grant tables and registers any unregistered privileges found.
+type FlushPrivileges struct {
+	flushType   string
```

Comment:
What is the `flushType`? Is that the `LOCAL` or `NO_WRITE_TO_BINLOG` option?

Reply:
I implemented all flush options and types specified in mysql Flush statement documentation. The types `LOCAL` or `NO_WRITE_TO_BINLOG` are not part of the flush-options; they specify whether the flush statement will be written into the binary log. The default is the flush statements are written to the binary log according to the mysql (docs)[https://dev.mysql.com/doc/refman/8.0/en/flush.html]

Reply:
I'm not sure I understand. This isn't for all of the `FLUSH` types, just `FLUSH PRIVILEGES`. `FLUSH PRIVILEGES` have two options that are prefixed before the `PRIVILEGES` keyword, giving the following variants:
* `FLUSH PRIVILEGES`
* `FLUSH NO_WRITE_TO_BINLOG PRIVILEGES`
* `FLUSH LOCAL PRIVILEGES`
Is the `flushType` variable specifying which variant we are using? If it is not, then it seems like an unused variable that should be removed.